storage/interceptable_repository.go

```diff
@@ -372,58 +327,34 @@ func (itr *InterceptableTransactionalRepository) Delete(ctx context.Context, obj
 	return objectList, err
 }
 
-type finalUpdateObjectInterceptor struct {
-	repository                 TransactionalRepository
-	objectType                 types.ObjectType
-	providedCreateInterceptors map[types.ObjectType]CreateInterceptor
-	providedUpdateInterceptors map[types.ObjectType]UpdateInterceptor
-	providedDeleteInterceptors map[types.ObjectType]DeleteInterceptor
-	criteria                   []query.Criterion
-}
-
-func (final *finalUpdateObjectInterceptor) InterceptUpdateOnTxFunc(ctx context.Context, txStorage Repository, obj types.Object, labelChanges ...*query.LabelChange) (types.Object, error) {
-	return txStorage.Update(ctx, obj, labelChanges)
-}
+func (itr *InterceptableTransactionalRepository) Update(ctx context.Context, obj types.Object, labelChanges query.LabelChanges, criteria ...query.Criterion) (types.Object, error) {
+	providedCreateInterceptors, providedUpdateInterceptors, providedDeleteInterceptors := itr.provideInterceptors()
 
-func (final *finalUpdateObjectInterceptor) InterceptUpdateAroundTxFunc(ctx context.Context, obj types.Object, labelChanges ...*query.LabelChange) (types.Object, error) {
-	var err error
-	var result types.Object
+	finalInterceptor := func(ctx context.Context, obj types.Object, labelChanges ...*query.LabelChange) (types.Object, error) {
+		var result types.Object
+		var err error
 
-	if err = final.repository.InTransaction(ctx, func(ctx context.Context, txStorage Repository) error {
-		interceptableRepository := newInterceptableRepository(txStorage, final.providedCreateInterceptors, final.providedUpdateInterceptors, final.providedDeleteInterceptors)
+		if err = itr.smStorageRepository.InTransaction(ctx, func(ctx context.Context, txStorage Repository) error {
+			interceptableRepository := newInterceptableRepository(txStorage, providedCreateInterceptors, providedUpdateInterceptors, providedDeleteInterceptors)
+			result, err = interceptableRepository.Update(ctx, obj, labelChanges)
```

Comment:
I think that the criteria from the closure should also be passed here (we should double check)

Reply:
It seems that the criteria has been forgotten way back and hasn't been passed on from the controller level as well - https://github.com/Peripli/service-manager/blob/7920247e060da10cb2f8ea7dced2f99823ae9c01/api/base_controller.go#L296 

Even if we pass it everywhere it is currently not in use on the lowest level (Storage implementation) - https://github.com/Peripli/service-manager/blob/7920247e060da10cb2f8ea7dced2f99823ae9c01/storage/postgres/storage.go#L258

I think we can start passing it everywhere where it's missing and on the lowest level (Storage implementation) mark it as unused by replacing the var name with underscore.

Reply:
Another idea would be to discard the criteria argument from the interface all together.